### DISCUSSION TITLE
I want to know current project's Baseline profile skill

### DISCUSSION THREAD

**Discussion by Jaehwa-Noh** [android/nowinandroid#1469] [Q&A] [answered]
Hello, @mlykotom
I saw your [video](https://youtu.be/WWdDzXgrkmg?feature=shared&t=547) and android [documentation](https://developer.android.com/topic/performance/baselineprofiles/create-baselineprofile#generate-profile) to study Baseline Profile.
The video's profile made output just three, and can easily pick up and rename `baseline-prof.txt`.
But, this project makes bunch of 11 outputs(include startup-prof.txt). as shown below image.
<img width="765" alt="image" src="https://github.com/android/nowinandroid/assets/48680511/fd2a0ebe-6238-4a5f-9a61-8cc6bd8f6492">

Current project serves the one `baseline-prof.txt` in src directory already made(first made by @keyboardsurfer)
I wonder how do you make those 11 files into one `baseline-prof.txt`.
Could you introduce about this?
Anyone involve in also welcome!

Thank you.

**Comment by mlykotom** [comment]
There's been an improvement in how baseline profiles are processed. Starting with AGP 7.2 you can have multiple files in the baselineProfile/ folder. These are then merged during app build.

This is why you have multiple files generated by the Macrobenchmark, because they're generated per your generator test.

You can copy the ones without the date postfix, it's there to provide unique file for test history, but it's equal to the ones without it.

Sorry for the confusion with the older setup in the videos!

**Reply by Jaehwa-Noh** [reply]
Thank you for answering.
Let me clear what I understand and add some additional questions.

1. In this projects, I just generate the Baseline Profile by run `:benchmarks:pixel6Api33DemoBenchmarkReleaseAndroidTest --rerun --enable-display -Pandroid.testInstrumentationRunnerArguments.androidx.benchmark.enabledRules=BaselineProfile`.
And no need to move and rename it to app/src directory because it will merged during app build.
> Starting with AGP 7.2 you can have multiple files in the baselineProfile/ folder. These are then merged during app build.

<br><br>

2. If no more need profiles move to src, in this project baseline profile at src no more need to remain so that can remove it?

<br><br>

3. The generated multiple profiles used in Macrobenchmark by `androidxProfileinstaller` as it's performance  testing.
> This is why you have multiple files generated by the Macrobenchmark, because they're generated per your generator test.

<br><br>

4. Why do I need manually copy if the build automatically merges the multiple baseline profiles? What cases do I need to copy this manually?
> You can copy the ones without the date postfix, it's there to provide unique file for test history, but it's equal to the ones without it.


<br><br>
5. This project run `:benchmarks:pixel6Api33DemoBenchmarkReleaseAndroidTest` to generate baseline profiles. Is there any different between `:benchmarks:pixel6Api33DemoBenchmarkReleaseAndroidTest` and `:app:pixel6Api33DemoBenchmarkReleaseAndroidTest`?

<br><br>
6. I found the results baseline profiles in `benchmarks/build/intermediates/managed_device_android_test_additional_output/demoBenchmarkRelease/pixel6Api33DemoBenchmarkReleaseAndroidTest/`. Do I found output path correctly? [documentation](https://developer.android.com/topic/performance/baselineprofiles/create-baselineprofile#generate-profile) says below but I can't found profiles that in path.
> When the Generate Baseline Profile run configuration completes, it copies the generated Baseline Profile to the src/variant/generated/baselineProfiles/baseline-prof.txt

<br><br>
7. Is startup-prof.txt also merged into baseline-prof.txt as it is building?
<br><br>

Sorry for lots of questions and clarifications.
Is there anything I do misunderstand?
Thank you.

**Reply by Jaehwa-Noh** [reply]
I would be confused by this project using `:benchmark:pixel6Api31atdDemoBenchmarkAndroidTest`.
When I'd switched it to  `:app:generateDemoReleaseBaselineProfile`, It automatically generated just one baseline and startup file in src.
#1468

**Reply by mlykotom** [reply, marked as answer]
If you run `xxxxBenchmarkReleaseAndroidTest` tasks, the generated baseline profile files **aren't** automatically moved to your project and you have to do it manually into the `src/main/baselineProfiles` folder. 
You can automate that step with the Baseline Profile Gradle Plugin and then you'd run `generateBaselineProfile` gradle task.
Check our [documentation](https://developer.android.com/topic/performance/baselineprofiles/create-baselineprofile).

>  6. I found the results baseline profiles in `benchmarks/build/intermediates/managed_device_android_test_additional_output/demoBenchmarkRelease/pixel6Api33DemoBenchmarkReleaseAndroidTest/`. Do I found output path correctly?

Sounds good, should be in `benchmarks/build/outputs` as well.

> Is startup-prof.txt also merged into baseline-prof.txt as it is building?
Startup-prof.txt can only be one and has to be named this way. This is specifically to improve startup of your app. 
In general, the profiles aren't merged into the **txt** file, but during the release compilation of your app, Android Gradle Plugin merged the text versions into one **binary** baseline profile that is then part of your APK.

(I'm a bit confused with the citing and the additional questions, sorry if I haven't answered everything)

**Reply by Jaehwa-Noh** [reply]
I'm fine, I understand your point of view. That was my lack of English expression.
Thank you for answering.
